gridstatus/ercot.py

```diff
@@ -3658,3 +3664,89 @@ def _handle_indicative_lmp_by_settlement_point(self, df: pd.DataFrame):
                 "LMP",
             ]
         ]
+
+    @support_date_range(frequency="DAY_START")
+    def get_dam_total_energy_purchased(
+        self,
+        date: str | pd.Timestamp,
+        end: str | pd.Timestamp | None = None,
+        verbose: bool = False,
+    ) -> pd.DataFrame:
+        """Get DAM Total Energy Purchased
+
+        Arguments:
+            date (str, datetime): date to get data for
+            end (str, datetime): end time to get data for
+            verbose (bool, optional): print verbose output. Defaults to False.
+
+        Returns:
+            pandas.DataFrame: A DataFrame with DAM total energy purchased data
+        """
+        if date == "latest":
+            return self.get_dam_total_energy_purchased(
+                date="today",
+                verbose=verbose,
+            )
+
+        # DAM data so subtract one from the date
+        doc = self._get_document(
+            report_type_id=DAM_TOTAL_ENERGY_PURCHASED_RTID,
+            date=date - pd.DateOffset(days=1),
+            extension="csv",
+        )
+
+        data = (
```

Comment:
i think we are going to want a `_handle_dam_total_energy_purchased_sold(doc)` method, because when we receive the zip of backfill files from ERCOT, we will want be able to call just those steps on them

Reply:
Makes sense. I'll wait until we have the files to implement.